bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerBaseHandler.java

```diff
@@ -0,0 +1,66 @@
+package org.openhab.binding.opensprinkler.internal.handler;
+
+import static org.openhab.binding.opensprinkler.internal.OpenSprinklerBindingConstants.DEFAULT_WAIT_BEFORE_INITIAL_REFRESH;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApi;
+
+public abstract class OpenSprinklerBaseHandler extends BaseThingHandler {
+    public OpenSprinklerBaseHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        OpenSprinklerApi ondusService = getApi();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
```

Comment:
If you are sure that `getApi()` is non-null when the bridge is online, you could move this to `bridgeStatusChanged`. Also better for recovery if the bridge goes offline in between and a new API needs to be requested.

Anyway: The `ThingStatusDetail` is wrong and needs to be changed. `BRIDGE_OFFLINE` should not be set by things. Maybe `COMMUNICATION_ERROR`?

Reply:
Hmm, good idea!

I don't think that we need to handle the "BRIDGE_OFFLINE" case, as the `BaseThingHandler` takes care of that, isn't it? So here we just need to handle the new bridge status ONLINE.

Reply:
The base thing handler is not re-initializating when the bridge comes online again. So if the Api potentially changes, you need to take care of that in your own implementation of `bridgeStatusChanged`.

Reply:
Hmm, is your comment based on the initialize implementation? It is already changed to use bridgeStatusChanged instead :/ Or do I just don't see what you're mentioning? As I understood it, this is already done in the implementation of bridgeStatusChanged? :)